### DISCUSSION TITLE
Is there a way to reference a container from another container deployed through kamal?

### DISCUSSION THREAD

**Discussion by AlexxPiP** [basecamp/kamal#731] [General]
If i have two services both deployed on the same machine through kamal, is there a way to have a request between the containers?
I noticed that both the name and hostname of the container are generated at build time and couldn't find any way to affect this behaviour. Beside these options,  the only way left to reference another container is through the container ip, but that is ephemeral too. 

Any help is appreciated, thanks

**Comment by nickhammond** [comment]
@AlexHashgraph You'll want to take a look at Docker networks. You can have your containers join the network and then you can use named hosts to communicate between them.

**Reply by AlexxPiP** [reply]
@nickhammond  but even if the containers are in the same network , after each kamal deploy the container will have a different hostname and name. 

If appA has a request to appB something like http://appB-hash. after a kamal deploy the host becomes http://appB-hash2, and service A requires an additional deploy with the new app B host.

**Reply by nickhammond** [reply]
I've only used it to point my app containers to an accessory(db, redis, elasticsearch, etc) container which does retain the same name "app-db" for instance. To point to your app containers you'd need to add some additional networking magic, Kamal doesn't help with this and isn't "meant" for single server deploys.

**Reply by AlexxPiP** [reply]
If I could control what kamal is passing when running the docker run, at least the hostname or container name , it would solve all my issues.

**Reply by AlexxPiP** [reply]
I found a way , by using the VERSION env variable, and set it to a static value on all deploys. This way i can reference from another container using the container name

**Reply by nickhammond** [reply]
Awesome, yah you can pass any additional options you need to docker in the config.

**Reply by dommmel** [reply]
> I found a way , by using the VERSION env variable, and set it to a static value on all deploys. This way i can reference from another container using the container name

Can you expand a bit on that? I am trying to achieve the same thing.
